examples/heartbeat_caravel_wrapper/build.py

```diff
@@ -0,0 +1,333 @@
+    jobdir = (core_chip.get('option', 'builddir') +
+            "/" + design + "/" +
+            core_chip.get('option', 'jobname'))
+    shutil.copy(f'{jobdir}/export/0/outputs/{design}.gds', f'{design}.gds')
+    shutil.copy(f'{jobdir}/export/0/inputs/{design}.def', f'{design}.def')
+    shutil.copy(f'{jobdir}/export/0/inputs/{design}.lef', f'{design}.lef')
+    shutil.copy(f'{jobdir}/dfm/0/outputs/{design}.vg', f'{design}.vg')
+
+def build_top():
+    # The 'hearbeat' RTL goes in a modified 'user_project_wrapper' object, see sources.
+    chip = configure_chip('user_project_wrapper')
+    chip.set('tool', 'openroad', 'var', 'place', '0', 'place_density', ['0.15'])
+    #chip.add('tool', 'openroad', 'var', 'place', '0', 'pad_global_place', ['2'])
+    #chip.add('tool', 'openroad', 'var', 'place', '0', 'pad_detail_place', ['2'])
+    chip.set('tool', 'openroad', 'var', 'route', '0', 'grt_allow_congestion', ['true'])
+    chip.clock('user_clock2', period=20)
+
+    # Set top-level source files.
+    chip.set('input', 'verilog', f'{CARAVEL_ROOT}/verilog/rtl/defines.v')
+    chip.add('input', 'verilog', 'heartbeat.bb.v')
+    chip.add('input', 'verilog', 'user_project_wrapper.v')
+
+    # Set top-level die/core area.
+    chip.set('asic', 'diearea', (0, 0))
+    chip.add('asic', 'diearea', (TOP_W, TOP_H))
+    chip.set('asic', 'corearea', (MARGIN_W, MARGIN_H))
+    chip.add('asic', 'corearea', (TOP_W - MARGIN_W, TOP_H - MARGIN_H))
+
+    # Add core design macro as a library.
+    libname = 'heartbeat'
+    stackup = chip.get('asic', 'stackup')
+    chip.add('asic', 'macrolib', libname)
+    heartbeat_lib = Chip('heartbeat')
+    heartbeat_lib.add('model', 'layout', 'lef', stackup, 'heartbeat.lef')
+    heartbeat_lib.add('model', 'layout', 'def', stackup, 'heartbeat.def')
+    heartbeat_lib.add('model', 'layout', 'gds', stackup, 'heartbeat.gds')
+    heartbeat_lib.add('model', 'layout', 'vg', stackup, 'heartbeat.vg')
+    heartbeat_lib.set('asic', 'pdk', 'skywater130')
+    heartbeat_lib.set('option', 'pdk', 'skywater130')
+    heartbeat_lib.set('asic', 'stackup', stackup)
+    chip.import_library(heartbeat_lib)
+
+    # Use pre-defined floorplan for the wrapper..
+    chip.set('input', 'floorplan.def', f'{CARAVEL_ROOT}/def/user_project_wrapper.def')
+
+    # (No?) filler cells in the top-level wrapper.
+    #chip.set('library', 'sky130hd', 'cells', 'filler', [])
+    #chip.add('library', 'sky130hd', 'cells', 'ignore', 'sky130_fd_sc_hd__conb_1')
+
+    # (No?) tapcells in the top-level wrapper.
+    libtype = 'unithd'
+    #chip.cfg['pdk']['aprtech']['openroad'][stackup][libtype].pop('tapcells')
+
+    # No I/O buffers in the top-level wrapper, but keep tie-hi/lo cells.
+    #chip.set('library', 'sky130hd', 'cells', 'tie', [])
+    chip.set('asic', 'cells', 'buf', [])
+
+    # Create PDN-generation script.
+    pdk = chip.get('option', 'pdk')
+    with open('pdngen_top.tcl', 'w') as pdnf:
```

Comment:
Yeah, the way things ended up, there isn't any design-specific information in this script file. So it might be a good idea to move this into a file of its own, but it would probably make sense to put that somewhere outside of the `examples/` directory, because it would be generic to the Caravel wrapper.

I'm not sure where the best place would be, and it would still require an extra schema entry defining the file's location.